src/olympia/devhub/tests/test_views_subscription.py

```diff
@@ -358,3 +358,61 @@ def test_get_does_not_set_payment_cancelled_date_when_already_paid(
         self.subscription.refresh_from_db()
 
         assert not self.subscription.payment_cancelled_at
+
+
+@override_switch("enable-subscriptions-for-promoted-addons", active=True)
+class TestSubscriptionCustomerPortal(SubscriptionTestCase):
+    url_name = "devhub.addons.subscription_customer_portal"
+
+    @override_switch("enable-subscriptions-for-promoted-addons", active=False)
+    def test_returns_404_when_switch_is_disabled(self):
+        assert self.client.post(self.url).status_code == 404
+
+    def test_allows_post_requests(self):
```

Comment:
This test name doesn't seem to match the contents. The test seems to be checking that the view disallows `get` requests, as opposed to allows `post` requests.

Reply:
Yeah, I see how it can be confusing. I'll update